\documentclass[11pt]{article}
\usepackage{fullpage}
\usepackage{titling}
\usepackage{listings}
\usepackage{xcolor}

\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}

\lstdefinestyle{mystyle}{
    backgroundcolor=\color{backcolour},   
    commentstyle=\color{codegreen},
    keywordstyle=\color{magenta},
    numberstyle=\tiny\color{codegray},
    stringstyle=\color{codepurple},
    basicstyle=\ttfamily\footnotesize,
    breakatwhitespace=false,         
    breaklines=true,                 
    captionpos=b,                    
    keepspaces=true,
    showspaces=false,                
    showstringspaces=false,
    showtabs=false,                  
    tabsize=2
}

\lstset{style=mystyle}

\setlength{\droptitle}{-5em}   % This is your set screw

\title{TreeFix-TP Manual}
% \author{
%     Samuel Sledzieski\\
%         Department of Computer Science\\
%         University of Connecticut\\
%     \and
%     Mukul Bansal \\
%         Department of Computer Science \\
%         University of Connecticut \\
% }
\date{}

\begin{document}
\maketitle
\vspace{-3cm}
\begin{center}
    Version 1.2.1, Compiled \today.
\end{center}

\section{Introduction}
TreeFix-TP is a program for reconstructing highly accurate transmission phylogenies, i.e., phylogenies depicting the evolutionary relationships between infectious disease strains (viral or bacterial) transmitted between different hosts. TreeFix-TP is designed for scenarios where multiple strain sequences have been sampled from each infected host, and it uses the host assignment of each sequence sample to error-correct a given maximum likelihood phylogeny of the strain sequences. Specifically, given a maximum likelihood phylogeny, the multiple sequence alignment on which the phylogeny was built, and the host assignment for each sequence, TreeFix-TP searches around the maximum likelihood phylogeny to find an alternate error-corrected phylogeny which is equally well-supported by the sequence data and minimizes the number of necessary inter-host transmissions.

\paragraph{Requirements}
    \begin{itemize}
        \item Python (3.5 or greater)
        \item C compiler (gcc)
        \item SWIG (1.3.29 or greater)
        \item Numpy (1.5.1 or greater)
        \item Scipy (0.7.1 or greater)
        \item Dendropy (Optional for \texttt{ttp-parse-log})
        \item Additionally, Python modules are required for computing the p-value for likelihood equivalence.
    \end{itemize}

\paragraph{Likelihood}
TreeFix-TP uses the Shimodaira-Hasegawa (SH) test statistic with RAxML site-wise likelihoods to compute
p-values for each candidate tree.

\paragraph{Parsimony}
TreeFix-TP scores each statistically equivalent candidate tree using Fitch's algorithm.
Given host labels at the leaf nodes, the Fitch module computes a score equivalent to the
minimum necessary number of transmissions needed to label the internal nodes.

\paragraph{Formatting Note} TreeFix-TP determines the host of each sequence by parsing the name assigned at the leaf. \textbf{Newick and Fasta formatted tree and sequence files should have the sequences in the form \texttt{[host name]\_[sequence name]}.} See \texttt{examples/test\_TP.fasta} for an example.

\section{Usage}

\paragraph{Input}
TreeFix-TP requires a seed tree, generally a maximum likelihood tree, and a multiple sequence alignment.

\paragraph{Options}
TreeFix-TP assumes that the multiple sequence alignment and seed tree are in the same directory with the same root name, with different extensions. The default extensions are ".fasta" and ".tree" for the multiple sequence alignment and maximum likelihood phylogeny, but other extensions can be specified. The output tree file will have the same root name, and will have either the default extension ".treefix.tree", or a user specified extension. The default statistical test (Shimodaira-Hasegawa) and cost calculation (Fitch's Algorithm) can be substituted with user defined Stat and Cost models.
\\

\begin{lstlisting}[language=Bash]
$ treefix-tp --help
Usage: treefix-tp [options] <gene tree> ...

TreeFix-TP is a phylogenetic program for improving viral phylogenetic tree
reconstructions using a test statistic for likelihood equivalence and a
transmission aware cost function. See http://github.com/samsledje/TreeFix-TP
for details.

Options:
  Input/Output:
    -A <alignment file extension>, --alignext=<alignment file extension>
                        alignment file extension (default: ".fasta")
    -o <old tree file extension>, --oldext=<old tree file extension>
                        old tree file extension (default: ".tree")
    -n <new tree file extension>, --newext=<new tree file extension>
                        new tree file extension (default: ".treefix.tree")
    -r, --reroot        set to reroot the input tree

  Likelihood Model:
    -m <module for likelihood calculations>, --module=<module for likelihood calculations>
                        module for likelihood calculations (default:
                        "treefix_tp.models.raxmlmodel.RAxMLModel")
    -e <extra arguments to module>, --extra=<extra arguments to module>
                        extra arguments to pass to program

  Likelihood Test:
    -t <test statistic>, --test=<test statistic>
                        test statistic for likelihood equivalence (default:
                        "SH")
    --alpha=<alpha>     alpha threshold (default: 0.05)
    -p <alpha>, --pval=<alpha>
                        same as --alpha

  Transmission Cost Model:
    -M <module for transmission cost calculation>, --smodule=<module for transmission cost calculation>
                        module for transmission cost calculation (default:
                        "treefix_tp.models.fitchmodel.FitchModel")

  Search Options:
    -b <# bootstraps>, --boot=<# bootstraps>
                        number of bootstraps to perform (default: 1)
    -x <seed>, --seed=<seed>
                        seed value for random generator
    --niter=<# iterations>
                        number of iterations (default: 100)
    --nquickiter=<# quick iterations>
                        number of subproposals (default: 50)
    --freconroot=<fraction reconroot>
                        fraction of search proposals to reconroot (default:
                        0.05)
    --maxtime=<maximum runtime>
                        maximum runtime (per tree) in seconds

  Information:
    --version           show program's version number and exit
    -h, --help          show this help message and exit
    -V <verbosity level>, --verbose=<verbosity level>
                        verbosity level (0=quiet, 1=low, 2=medium, 3=high)
    -l <log file>, --log=<log file>
                        log filename.  Use '-' to display on stdout.

  Debug:
    --debug=<debug mode>
                        debug mode (octal: 0=normal, 1=skips likelihood test,
                        2=skips cost filtering on pool, 4=computes likelihood
                        for all trees in pool)
\end{lstlisting}
\vspace{2mm}

\subsection{Additional Tools}
\paragraph{ttp-parse-log}

Get summary statistics and consensus trees from the phylogenies considered by TreeFix-TP  (requires DendroPy).

\begin{lstlisting}[language=Bash]
$ ttp-parse-log --help
Usage: ttp-parse-log [options] <log file>

Options:
  -h, --help            show this help message and exit
  --out=OUT_PATH        Path for output
  --near=NEAR_PERCENT   Trees within <--near>% of the optimal cost will be
                        captured
  --true=TRUE_TREE_PATH
                        Can provide a true tree to compare multiple optimal
                        trees with
  --include_near        Include nearby optimal trees in summary statistics
  --separate_trees      Create two output files separating trees and
                        statistics
\end{lstlisting}
\vspace{2mm}

\paragraph{ttp-check-cost}
Compare the transmission cost of the input phylogeny and the TreeFix-TP optimal phylogeny.

\begin{lstlisting}[language=Bash]
$ ttp-check-cost --help
usage: python ttp-check-cost.py [old tree file] [new tree file]
\end{lstlisting}
\vspace{2mm}

\paragraph{ttp-check-likelihood}
Compare the sequence support of the input phylogeny with the TreeFix-TP optimal phylogeny.

\begin{lstlisting}[language=Bash]
$ ttp-check-likelihood --help
usage: python ttp-check-likelihood.py [alignment] [old tree file] [new tree file]
\end{lstlisting}
\vspace{2mm}

\section{Example}

\begin{lstlisting}[language=Bash]
$ cd examples/
$ # Show Files
$ ls
test.sh    test_TP.fasta    test_TP.network    test_TP.raxml    test_TP.true_tree

$ # Run TreeFix-TP
$ treefix-tp -A .fasta -o .raxml -n .treefix -V1 -l test_TP.log test_TP.raxml

$ # Show Consensus Trees and compare with true tree
$ ttp-parse-log test_TP.log --true test_TP.true_tree

$ # Check Sequence Support
$ ttp-check-likelihood test_TP.raxml test_TP.treefix

$ # Check Cost Decrease
$ ttp-check-cost test_TP.fasta test_TP.raxml test_TP.treefix
\end{lstlisting}
\vspace{2mm}

\section*{Attribution}
Written by Samuel Sledzieski (samuel.sledzieski@uconn.edu) and Mukul Bansal
(mukul.bansal@uconn.edu), University of Connecticut. (c) 2020. Released under
the terms of the GNU General Public License.
Treefix written by Yi-Chieh Wu (yjw@mit.edu), Massachusetts Institute of
Technology. (c) 2011. Released under the terms of the GNU General Public
License. If you use TreeFix-TP please cite\\
\texttt{TreeFix-TP: Phylogenetic Error-Correction for Infectious Disease Transmission Network Inference. Samuel Sledzieski, Chengchen Zhang, Ion Mandoiu, and Mukul S. Bansal. Pacific Symposium on Biocomputing (PSB) 2021: Proceedings, pages 119-130.}

% \section{Changes from TreeFix-DTL}
% TreeFix-TP was built by modifying TreeFix-DTL \cite{treefix-dtl}, with the most significant change being the replaced cost module (fitch.linux rather than ranger-dtl). All changes are documented below.

% \subsection{treefixTP}
% \begin{itemize}
%     \item Based on treefixDTL script
%     \item Removed most options in parser, only need alignment extension, old extension, new extension
%     \item Changed default smodule to FitchModel
%     \item Call treefix{\_}for{\_}TP rather than treefix
% \end{itemize}

% \subsection{fitchmodel.py}
% \begin{itemize}
%     \item Copy of rangerdtlmodel.py
%     \item Changed compute{\_}cost() to call fitch.linux executable
%     \item Removed stree, smap from FitchModel.optimize{\_}model
%     \item Removed call to CostModel.optimize{\_}model()
% \end{itemize}

% \subsection{treefix{\_}for{\_}TP}
% \begin{itemize}
%     \item Renamed treefix to treefix{\_}for{\_}TP - this avoids path collision with treefix and treefixDTL
%     \item Removed smap and stree as required arguments
%     \item Changed to common.check{\_}req{\_}options(species=False)
%     \item Removed stree and gene2species from check{\_}input{\_}tree()
%     \item Removed stree and gene2species from search{\_}landscape()
%     \item Removed check{\_}congruent{\_}tree() function, no longer applies
%     \item Removed `if flag: return mintree'
%     \item Removed reading species tree and species map from main()
%     \item Removed stree and gene2species from calls to optimize{\_}model(), check{\_}input{\_}tree(), and search{\_}landscape() in main
% \end{itemize}

% \subsection{General}
% \begin{itemize}
%     \item Removed all ranger executables
%     \item Removed treefix{\_}compute
%     \item Removed rangerdtl model from models module
%     \item Modified setup files to install the proper executables
%     \item Updated treefixTP.py with version and software info
%     \item Updated INSTALL.txt, CHANGES.txt, and README.txt
%     \item Added README.pdf
% \end{itemize}

\bibliographystyle{plain}
\end{document}